src/encryptsave/mod.rs

```diff
@@ -0,0 +1,175 @@
+mod ll;
+pub mod errors;
+
+
+macro_rules! tox_res {
+    ( $res:ident <- $rexpr:expr, $err:ident, $r:expr ) => {
```

Comment:
Yep, I see. Well, it can be `( $res:ident, $err:ident, $e:expr )` then. Something like “`Ok($res)` if no error, `Err($err)` otherwise”. Additional conditional is rather confusing than helping.